tests/RouteTest.php

```diff
@@ -169,46 +170,155 @@ static function (): ResponseInterface {
         $this->assertSame(418, $response->getStatusCode());
     }
 
-    public function testThen(): void
+    public function testThenByStatusCode(): void
     {
         $request = new ServerRequest('GET', '/');
 
-        $route = Route::get('/');
+        $routeOne = Route::get('/');
+
+        $middleware1 = new Callback(function (ServerRequestInterface $request, RequestHandlerInterface $handler) {
+            return $handler->handle($request);
+        });
+        $middleware2 = new Callback(function (ServerRequestInterface $request, RequestHandlerInterface $handler) {
+            return new Response(200);
+        });
 
-        $middleware1 = $this->createMock(MiddlewareInterface::class);
-        $middleware2 = $this->createMock(MiddlewareInterface::class);
+        $routeOne = $routeOne->to($middleware1)->then($middleware2);
 
-        $route = $route->to($middleware1)->then($middleware2);
+        $response = $routeOne->process($request, $this->getRequestHandler());
+        $this->assertSame(200, $response->getStatusCode());
 
-        $middleware1
-            ->expects($this->at(0))
-            ->method('process')
-            ->with($request, $route);
+        $routeTwo = Route::get('/');
 
-        // TODO: test that second one is called as well
+        $middleware1 = new Callback(function (ServerRequestInterface $request, RequestHandlerInterface $handler) {
+            return new Response(404);
+        });
+        $middleware2 = new Callback(function (ServerRequestInterface $request, RequestHandlerInterface $handler) {
+            return new Response(200);
+        });
 
-        $route->process($request, $this->getRequestHandler());
+        $routeTwo = $routeTwo->to($middleware1)->then($middleware2);
+
+        $response = $routeTwo->process($request, $this->getRequestHandler());
+        $this->assertSame(404, $response->getStatusCode());
     }
 
-    public function testBefore(): void
+    public function testThenByBody(): void
```

Comment:
This test is similar to `testThenByStatusCode`. It has two tests and these are checking same things: execution of the whole middleware stack and ability to stop stack processing. Is there a need for duplicating the test?

Reply:
No need, just historically it was done during the development process. Which of them will we leave?

Reply:
Statuses, I think. They're a bit easier to read.

Reply:
> Statuses, I think. They're a bit easier to read.

I agree, moreover it is faster.

Reply:
Done.